### ISSUE TITLE
Ability to move the local video preview around a 1:1 call

### ISSUE THREAD

**Issue by ara4n** [vector-im/element-web#18262]
It's frustrating that you can move the PIP around, but not the local video preview

**Comment by Palid** [comment]
Related: https://github.com/matrix-org/matrix-react-sdk/pull/6475

**Comment by SimonBrandner** [comment]
> Related: [matrix-org/matrix-react-sdk#6475](https://github.com/matrix-org/matrix-react-sdk/pull/6475)

As far as I can tell, this is talking about the view of yourself inside the call view, so I don't think your PR has much to do with that, or am I missing something?

**Comment by Palid** [comment]
@SimonBrandner yeah right, I misunderstood reading that for the first time it seems, though I think it may still be related as the solution will be very similar, or even exactly the same.

**Comment by SimonBrandner** [comment]
Well, not really, with https://github.com/matrix-org/matrix-react-sdk/pull/5992 there can be multiple secondary feeds at the same time, so this will need some design thinking

**Comment by SimonBrandner** [comment]
@gaelledel, do you have any thoughts on this? I have a certain preference for making the secondary feeds behave like a sidebar (like Discord does), which would solve this issue, IMO. I have an awfully unprofessional Figma file [here](https://www.figma.com/file/Q1ZXFyyZuGGgqD4PRzXwmr/%5BConcept%5D-VoIP-Sidebar?node-id=3472%3A78779) but perhaps there are other solutions

**Comment by ara4n** [comment]
I think we're getting a little off course here - i'm just asking to be able to drag around your local video preview (of which there is only one) within the overall video frame.  Currently it's wedged in the top right, whereas I might want to move it somewhere to be less distracting (or hide it entirely).

I don't think this needs design, and i don't think it's related to the separate problem of how you visualise multiple remote video feeds (which @gaelledel is separately looking at in the context of full mesh video calling)

**Comment by SimonBrandner** [comment]
It is a design issue because there can be multiple feeds - 2 usermedia ones and 2 screen sharing ones. I.e. one primary and  max 3 secondaries atm

**Comment by SimonBrandner** [comment]
Also, we have a button to hide the secondaries, if there is more than one.

(all of this is coming from: https://github.com/matrix-org/matrix-react-sdk/pull/5992)

**Comment by Palid** [comment]
@ara4n this is a non-issue if we consider it working only for one feed, but then what the behavior should be when we go to multiple feeds from one (e.g. screen sharing + camera)? Should you be able to move all of them around? Should it just snap to one corner instead? 
It would create a regressive experience if we don't account for it with multiple feeds, or even worse, make it confusing by default, so IMHO we need to figure out the design and UX first for the harder cases. 

I think that it'll be much easier to account for "good enough" user experience if we first figure out multi-feed, as @SimonBrandner stated above, so that we won't make it confusing going up with the amount of streams. 

Maybe the temporary solution would be to just make the button hiding preview work for **all** of your streams, not only secondary ones? This won't be a very impactful on the future UX methinks.

**Comment by SimonBrandner** [comment]
> Maybe the temporary solution would be to just make the button hiding preview work for all of your streams, not only secondary ones? This won't be a very impactful on the future UX methinks.

We could make the button _always_ visible, not only when screen-sharing, which would at least partially solve the issue

**Comment by Nama** [comment]
Is this implemented now? I have a preview of the call if I change the room from the call and can move it around.

But there seems to be a bug. The preview goes back to bottom right after visiting the room the call is in and clicking on a different room again. It should always stay at the position I moved it to.

Ability to hide it until I enable it again would be nice, too.
Thanks for working on this!

**Comment by SimonBrandner** [comment]
This issues speaks about the possibility of moving the **local** video (the one of yourself) inside a 1:1 call which is not implemented, afaik. You are talking about the Picture in Picture (PiP) view of the call from other rooms. The issue you mention is https://github.com/element-hq/element-web/issues/18589
